ts/src/renderer/phaser/classes/PhaserEntity.ts

```diff
@@ -0,0 +1,61 @@
+abstract class PhaserEntity extends Phaser.GameObjects.Container {
+
+	sprite: Phaser.GameObjects.Sprite;
+    key: any;
+
+	private playAnimationListener: EvtListener;
+	private transformListener: EvtListener;
+	private scaleListener: EvtListener;
+	private destroyListener: EvtListener;
+
+	constructor (scene: Phaser.Scene,
+				 private entity: IgeEntity) {
+
+		super(scene);
+
+		//const key = `projectile/${entity._stats.type}`;
+
+		const sprite = this.sprite = scene.add.sprite(0, 0, null);
+		const translate = entity._translate;
+		//const bounds = entity._bounds2d;
+		this.setPosition(translate.x, translate.y);
+		sprite.rotation = entity._rotate.z;
+		//sprite.setDisplaySize(bounds.x, bounds.y);
+
+		this.add(sprite);
+
+		scene.add.existing(this);
+
+		this.transformListener = entity.on('transform', (data: {
```

Comment:
I suggest defining event listeners as class methods. that way sub-classes can extend or override basic behavior without needing to reassign event listeners.
for example have here: 
`this.transformListener = entity.on('transform', this.transform, this);`
and define transform method below as:
```
transform (data: {
    x: number,
	y: number,
	rotation: number
}): void => {
	this.setPosition(data.x, data.y);
	this.sprite.rotation = data.rotation;
=});